pkgs/development/tools/database/shmig/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchFromGitHub
+, withMySQL ? false, withPSQL ? false, withSQLite ? false
+, mariadb, postgresql, sqlite, gawk, which
+}:
+
+stdenv.mkDerivation {
+  name = "shmig-2017-07-24";
+
+  src = fetchFromGitHub {
+    owner = "mbucc";
+    repo = "shmig";
+    rev = "aff54e03d13f8f95b422cf898505490a56152a4a";
+    sha256 = "08q94dka5yqkzkis3w7j1q8kc7d3kk7mb2drx8ms59jcqvp847j3";
+  };
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  postPatch = let
+    replacePath = cond: replacement:
+      if cond then "\"${replacement}\""
+      else "";
+  in ''
+    patchShebangs .
+
+    substituteInPlace shmig \
+      --replace "\`which mysql\`" "${replacePath withMySQL "${mariadb}/bin/mysql"}" \
+      --replace "\`which psql\`" "${replacePath withPSQL "${postgresql}/bin/psql"}" \
+      --replace "\`which sqlite3\`" "${replacePath withSQLite "${sqlite}/bin/sqlite3"}" \
+      --replace "awk" "${gawk}/bin/awk" \
+      --replace "which" "${which}/bin/which"
+  '';
+
+  preBuild = ''
+    mkdir -p $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Database migration tool written in BASH.
```

Comment:
Just a nit but is the implementation language really important information?

Reply:
I always use the description from GitHub. However I think that you might be right, I'll fine a better one :-)

Reply:
I don't consider it a blocker